tests/cbtest/gtest/test_uzfs.cc

```diff
@@ -773,12 +790,206 @@ TEST(uZFS, TestIOConnAcceptor) {
+		rc = uzfs_zvol_socket_write(sfd, (char *)&hdr, sizeof (hdr));
+		if (rc != 0) {
+			LOG_ERRNO("Socket rebuild_complete write failed, but,"
+			    "counting as success with this replica");
+			rc = 0;
+			goto exit;
+		}
+
+		rc = 0;
+		LOG_INFO("Rebuilding zvol %s completed", zinfo->name);
+		goto exit;
+	}
+
+	while (1) {
+
+		rc = uzfs_zvol_socket_read(sfd, (char *)&hdr, sizeof (hdr));
+		if (rc != 0) {
+			LOG_ERR("Socket read failed");
+			goto exit;
+		}
+
+		if (hdr.status != ZVOL_OP_STATUS_OK) {
+			LOG_ERR("received err in rebuild.. for %s..",
+			    zinfo->name);
+			rc = -1;
+			goto exit;
+		}
+
+		if (hdr.opcode == ZVOL_OPCODE_REBUILD_STEP_DONE) {
+			offset += zvol_rebuild_step_size;
+			LOG_DEBUG("ZVOL_OPCODE_REBUILD_STEP_DONE received");
+			goto next_step;
+		}
+
+		ASSERT((hdr.opcode == ZVOL_OPCODE_READ) &&
+		    (hdr.flags & ZVOL_OP_FLAG_REBUILD));
+		hdr.opcode = ZVOL_OPCODE_WRITE;
+
+		zio_cmd = zio_cmd_alloc(&hdr, sfd);
+		rc = uzfs_zvol_socket_read(sfd, (char *)zio_cmd->buf, hdr.len);
+		if (rc != 0) {
+			LOG_ERR("Socket read writeIO failed");
+			goto exit;
+		}
+
+		/*
+		 * Take refcount for uzfs_zvol_worker to work on it.
+		 * Will dropped by uzfs_zvol_worker once cmd is executed.
+		 */
+		uzfs_zinfo_take_refcnt(zinfo, B_FALSE);
+		zio_cmd->zv = zinfo;
+		uzfs_zvol_worker(zio_cmd);
+		if (zio_cmd->hdr.status != ZVOL_OP_STATUS_OK) {
+			LOG_ERR("rebuild IO failed.. for %s..", zinfo->name);
+			rc = -1;
+			goto exit;
+		}
+		zio_cmd_free(&zio_cmd);
+	}
+
+exit:
+	mutex_enter(&zinfo->zv->rebuild_mtx);
+	if (rc != 0) {
+		uzfs_zvol_set_rebuild_status(zinfo->zv,
+		    ZVOL_REBUILDING_ERRORED);
+		(zinfo->zv->rebuild_info.rebuild_failed_cnt) += 1;
+		LOG_ERR("uzfs_zvol_rebuild_dw_replica thread exiting, "
+		    "rebuilding failed zvol: %s", zinfo->name);
+	}
+	(zinfo->zv->rebuild_info.rebuild_done_cnt) += 1;
+	if (zinfo->zv->rebuild_info.rebuild_cnt ==
+	    zinfo->zv->rebuild_info.rebuild_done_cnt) {
+		if (zinfo->zv->rebuild_info.rebuild_failed_cnt != 0)
+			uzfs_zvol_set_rebuild_status(zinfo->zv,
+			    ZVOL_REBUILDING_FAILED);
+		else {
+			/* Mark replica healthy now */
+			uzfs_zvol_set_rebuild_status(zinfo->zv,
+			    ZVOL_REBUILDING_DONE);
+			uzfs_zvol_set_status(zinfo->zv, ZVOL_STATUS_HEALTHY);
+			uzfs_update_ionum_interval(zinfo, 0);
+		}
+	}
+	mutex_exit(&zinfo->zv->rebuild_mtx);
+
+	kmem_free(arg, sizeof (rebuild_thread_arg_t));
+	if (zio_cmd != NULL)
+		zio_cmd_free(&zio_cmd);
+	if (sfd != -1) {
+		shutdown(sfd, SHUT_RDWR);
+		close(sfd);
+	}
+	/* Parent thread have taken refcount, drop it now */
+	uzfs_zinfo_drop_refcnt(zinfo, B_FALSE);
+
+	rebuild_test_case = 0;
+	zk_thread_exit();
+}
+
+void execute_rebuild_test_case(const char *s, int test_case,
+    zvol_rebuild_status_t status, boolean_t is_test_positive,
+    boolean_t is_rebuild_scanner)
 {
 	kthread_t *thrd;
 	rebuild_thread_arg_t *rebuild_args;
 
-	printf("%s\n", s);
+	printf("========%s========\n", s);
```

Comment:
This can be removed now as the test suite will print the name of the test which is running. In fact the argument `s` should be removed entirely.

Reply:
done